File: src/models/relation.py
from datetime import datetime

from pydantic import BaseModel, Field

from src.models.enums import RelationStatus


class Relation(BaseModel):
    id: str
    world_id: str
    character_a: str
    character_b: str
    type: str | None = None
    direction: str = "bidirectional"
    description: str | None = None
    status: str = RelationStatus.ACTIVE
    historical_changes: list[dict] | None = None
    graph_edge_uuid: str | None = None
    created_at: datetime = Field(default_factory=datetime.now)
    updated_at: datetime = Field(default_factory=datetime.now)


class CreateRelationRequest(BaseModel):
    character_a: str = Field(min_length=1)
    character_b: str = Field(min_length=1)
    type: str | None = None
    direction: str = "bidirectional"
    description: str | None = None


class UpdateRelationRequest(BaseModel):
    type: str | None = None
    direction: str | None = None
    description: str | None = None
    status: str | None = None

File: frontend/src/types/relation.ts
export interface Relation {
  id: string
  world_id: string
  character_a: string
  character_b: string
  type: string | null
  direction: string
  description: string | null
  status: string
  historical_changes: Record<string, unknown>[] | null
  graph_edge_uuid?: string | null
  created_at: string
  updated_at: string
}

export interface CreateRelationRequest {
  character_a: string
  character_b: string
  type?: string | null
  direction?: string
  description?: string | null
}

export interface UpdateRelationRequest {
  type?: string | null
  direction?: string | null
  description?: string | null
  status?: string | null
}
